src/useAnswersState.tsx

```diff
@@ -16,57 +17,32 @@ export function useAnswersState<T>(stateSelector: StateSelector<T>): T {
      + ' Please ensure that \'useAnswersState()\' is called within an AnswersHeadlessProvider component.');
   }
 
-  // useRef stores values across renders without triggering additional ones
-  const storedStoreState = useRef<State>(answers.state);
-  const storedSelector = useRef<StateSelector<T>>(stateSelector);
-  const storedSelectedState = useRef<T>();
-  /**
-   * Guard execution of {@link stateSelector} for initializing storedSelectedState.
-   * Otherwise it's run an additional time every render, even when storedSelectedState is already initialized.
-   */
-  if (storedSelectedState.current === undefined) {
-    storedSelectedState.current = stateSelector(answers.state);
-  }
-
-  /**
-   * The currently selected state - this is the value returned by the hook.
-   * Tries to use {@link storedSelectedState} when possible.
-   */
-  const selectedStateToReturn: T = (() => {
-    if (storedStoreState.current !== answers.state || storedSelector.current !== stateSelector) {
-      return stateSelector(answers.state);
-    }
-    return storedSelectedState.current;
-  })();
-
-  const [, triggerRender] = useState<T>(storedSelectedState.current);
-  useLayoutEffect(() => {
-    storedSelector.current = stateSelector;
-    storedStoreState.current = answers.state;
-    storedSelectedState.current = selectedStateToReturn;
-  });
+  const getSnapshot = useCallback(() => answers.state, [answers.state]);
```

Comment:
What happens if this isn't wrapped in useCallback? totally fine if it's unknown don't need to deepdive just asking if you already knew. I guess I'm more concerned with if wrapping this in useCallback is the recommended usage?

Reply:
subscribe function and getSnapshot function are in the dependency array in the useLayoutEffect and useEffect calls in use-sync-external-store so having new function created every time would cause additional updates/renders if it's not wrapped in useCallback!